docs/source/tutorials/python_in_mantid/solutions/02_pim_sol.rst

```diff
@@ -0,0 +1,141 @@
+.. _02_pim_sol:
+
+============================
+Python in Mantid: Exercise 2
+============================
+
+A - Processing ISIS Data
+========================
+
+.. code-block:: python
+
+    # import mantid algorithms, numpy and matplotlib
+    from mantid.simpleapi import *
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    # Load the main data file and its monitor
+    Load(Filename='LOQ48097.raw', OutputWorkspace='Small_Angle', LoadMonitors='Separate')
+
+    # Convert monitor to wavelength
+    ConvertUnits(InputWorkspace='Small_Angle_monitors', OutputWorkspace='Small_Angle_monitors', Target='Wavelength')
+
+    # Convert data to wavelength
+    ConvertUnits(InputWorkspace='Small_Angle', OutputWorkspace='Small_Angle', Target='Wavelength')
+
+    # Rebin the monitors with a suggested set of parameters
+    rebin_var = Rebin(InputWorkspace='Small_Angle_monitors', OutputWorkspace='Small_Angle_monitors', Params='2.2,-0.035,10')
+
+    # Extract binning params from the first Rebin-algm
+    rebin_alg = rebin_var.getHistory().lastAlgorithm()
+    params = rebin_alg.getPropertyValue('Params')
+
+    # Log the Rebin params at the level information
+    logger.information("Rebin parameters: {}".format(params))
+
+    # Rebin the data with the params extracted from the earlier Rebin
+    Rebin(InputWorkspace='Small_Angle', OutputWorkspace='Small_Angle', Params= params)
+
+    # Extract the Spectrum for correcting the data
+    ExtractSingleSpectrum(InputWorkspace='Small_Angle_monitors', OutputWorkspace='Small_Angle_monitors', WorkspaceIndex=1)
+
+    # Correct the data by dividing by the monitor spectrum
+    Divide(LHSWorkspace='Small_Angle', RHSWorkspace='Small_Angle_monitors', OutputWorkspace='Corrected_data')
+
+
+B - Plotting ILL Data
```

Comment:
There’s two code blocks in this solution and only the python bit gets created in the html page, which starts with ADS.retrieve('164198), however at this point that workspace doesn't exist. So maybe the first call needs to be load.

Reply:
I understand that this "doesn't work" as it doesn't actually load the data, but exercise B was simply about generating a script to produce a plot. This will work for the people taking the course as they will already have the data loaded in the Workspaces Toolbox. I also want to display the plot so in the plot directive below this code-block I actually do load the data.